### PR TITLE
Add CacheLevel to AuthResultMetadata

### LINKED ISSUE #4304: [Feature Request] Expose Cache Level in AuthenticationResultMetadata
### MSAL client type

Confidential

### Problem Statement

We have made CacheLevel a public API but we do not expose it in AuthenticationResult for programatic consumption. 

### Proposed solution

_No response_

### Alternatives

_No response_

### PR DESCRIPTION
**Pull request by pmaytak**
Fixes #4304

**Changes proposed in this request**
Add `CacheLevel` to `AuthenticationResultMetadata`.

**Testing**
Updated unit test.

**Performance impact**
N/A

**Documentation**
- [x] All relevant documentation is updated.

### CODE REVIEW

File: tests/Microsoft.Identity.Test.Unit/PublicApiTests/TelemetryClientTests.cs [hunk 1]
```diff
@@ -494,11 +494,11 @@ public async Task AcquireTokenGenericErrorTelemetryTestAsync()
             Assert.AreEqual(authenticationResult.AuthenticationResultMetadata.DurationTotalInMs, eventDetails.Properties[TelemetryConstants.Duration]);
             Assert.AreEqual(authenticationResult.AuthenticationResultMetadata.DurationInHttpInMs, eventDetails.Properties[TelemetryConstants.DurationInHttp]);
             Assert.AreEqual(authenticationResult.AuthenticationResultMetadata.DurationInCacheInMs, eventDetails.Properties[TelemetryConstants.DurationInCache]);
-            Assert.AreEqual(authenticationResult.AuthenticationResultMetadata.DurationTotalInMs, eventDetails.Properties[TelemetryConstants.Duration]);
```
Comment (gladjohn) [inline tests/Microsoft.Identity.Test.Unit/PublicApiTests/TelemetryClientTests.cs:497]: why was this removed?

**Review by bgavrilMS — APPROVED** [review]

File: tests/Microsoft.Identity.Test.Unit/PublicApiTests/TelemetryClientTests.cs (hunk 1 above)
Comment (pmaytak) [reply tests/Microsoft.Identity.Test.Unit/PublicApiTests/TelemetryClientTests.cs:497]: Duplicate Line 494.

**Review by gladjohn — APPROVED** [review]
